### PR TITLE
Development review backend

### PR DESCRIPTION
**Pull request by rawdaGastan**
### Related review PR

- https://github.com/codescalers/cloud4students/pull/254

cmd + internal review

ps: new mail templates
Screenshot from 2023-05-22 15-13-59

### CODE REVIEW

File: server/internal/hash_password.go [hunk 1]
```diff
@@ -2,23 +2,32 @@
...
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"io"
 )
 
-// HashAndSaltPassword hashes password of user
-func HashAndSaltPassword(password string, salt string) (string, error) {
-	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(salt+password), bcrypt.MinCost)
+var saltLen = 16
```
Comment (xmonader) [inline server/internal/hash_password.go:11]: is this going to change?

File: server/internal/config_parser.go [hunk 2]
```diff
@@ -25,100 +25,50 @@ type Configuration struct {
...
-func ReadConfFile(path string) ([]byte, error) {
-	confFile, err := os.Open(path)
-	if err != nil {
-		return nil, err
-	}
-
-	defer confFile.Close()
-	conf, err := io.ReadAll(confFile)
+func ReadConfFile(path string) (Configuration, error) {
+	config := Configuration{NotifyAdminsIntervalHours: 6, BalanceThreshold: 2000}
```
Comment (xmonader) [inline server/internal/config_parser.go:62]: Not sure why these values exist here, or that's for default values?

File: server/internal/config_parser_test.go [hunk 3]
```diff
@@ -57,49 +78,26 @@ func TestParseConf(t *testing.T) {
 
 	t.Run("can't unmarshal", func(t *testing.T) {
 		config := `{testing}`
-		_, err := ParseConf([]byte(config))
+
+		dir := t.TempDir()
+		configPath := dir + "/config.json"
```
Comment (xmonader) [inline server/internal/config_parser_test.go:83]: please make sure to use filepath.join

File: server/validators/user_validator.go [hunk 4]
```diff
@@ -25,8 +25,7 @@ func ValidateMail(v interface{}, param string) error {
 	if st.Kind() != reflect.String {
 		return errors.New("ValidateMail only validates strings")
 	}
-	_, err := mail.ParseAddress(st.String())
-	return err
+	return ValidMail(st.String())
```
Comment (xmonader) [inline server/validators/user_validator.go:28]: Let's make all of the errors lowercased in `ValidateMail`, and `ValidateSSHKey`, .. etc

File: server/internal/config_parser.go (hunk 2 above)
Comment (rawdaGastan) [reply server/internal/config_parser.go:62]: they are the default ones

**Review by xmonader — APPROVED** [review]

**Pull request merged**